### DISCUSSION TITLE
Deleteing a property from an Immer Entity store doesn't work

### DISCUSSION THREAD

**Discussion by IAMtheIAM** [salesforce/akita#662] [Q&A]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

```

// Store config
@StoreConfig({
    name: 'my-store', producerFn: produce, resettable: true
})

// Delete property from stat
        this.store.update((state: StoreModel) => {
            delete state.myProp;
        });
```

## Expected behavior
state.myProp should no longer exist, however, it remains there unchanged.

## Minimal reproduction of the problem with instructions

Run above code, notice it doesn't delete the property.

## Environment

<pre><code>
Angular version: Vue 2.6.11

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version 85.0.2 (64-bit)
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by NetanelBasal** [comment]
Can you first try it by using plain immer?
